widget/list.go

```diff
@@ -458,25 +457,32 @@ func (l *listLayout) Layout(objects []fyne.CanvasObject, size fyne.Size) {
 		return
 	}
 	y := 0
-	for _, child := range objects {
+	for _, child := range l.children {
 		child.Move(fyne.NewPos(0, y))
-		y += l.list.itemMin.Height
+		y += l.list.itemMin.Height + listDividerHeight
 		child.Resize(fyne.NewSize(l.list.size.Width, l.list.itemMin.Height))
 	}
 	l.layoutEndY = y
+	l.updateDividers()
 }
 
 func (l *listLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
 	if f := l.list.Length; f != nil {
 		return fyne.NewSize(l.list.itemMin.Width,
-			l.list.itemMin.Height*f())
+			l.list.itemMin.Height*f()+(f()-1)*listDividerHeight)
```

Comment:
You could avoid calling f() twice, and remove the second multiplication using:

```suggestion
			(l.list.itemMin.Height+listDividerHeight)*f()-listDividerHeight
```